upgrade/upgrade-scripts/execute

```diff
@@ -612,7 +612,8 @@ fi
 if { [[ -n "$opt_f" ]] || [[ "$UPGRADE_TYPE" == "FULL" ]]; } &&
 	compare_versions "$CURRENT_VERSION" lt "18.0.0.0-0"; then
 	post_alert "reboot"
-	exec systemctl reboot || die "failed to reboot"
+	systemctl reboot || die "failed to reboot"
+	exec sleep infinity || die "failed to wait for the reboot"
```

Comment:
It might be good to ad a comment above the sleep explaining that it's there solely to prevent an issue when run from versions <= 16.0.0.0. Once the minimum upgrade version is > 16.0.0.0, this won't be needed anymore, right?

Reply:
yea, maybe.. although, I kind of like having the semantics be, this `execute` script won't return until the reboot occurs (i.e. it'll never return).. or the delphix services have been restarted..

I think that's what the code that consumes this script was assuming.. but since `systemctl reboot` returns prior to the reboot happening, that assumption was broken, leading to the bug..

in which case, I think a comment to explain the intention (i.e. sleep, to maintain these semantics), but not one that is specific to versions..

does that make sense?

Reply:
yeah, makes sense.